bleak/backends/bluezdbus/client.py

```diff
@@ -53,12 +53,14 @@ def __init__(self, address_or_ble_device: Union[BLEDevice, str], **kwargs):
         self._adapter: Optional[str] = kwargs.get("adapter", kwargs.get("device"))
 
         # Backend specific, D-Bus objects and data
-        if isinstance(address_or_ble_device, BLEDevice):
+        self._device_path = None
+        self._device_info = None
+        if (
+            isinstance(address_or_ble_device, BLEDevice)
+            and address_or_ble_device.details
+        ):
```

Comment:
Sure, no problems I will drop this commit.

FWIW, I added the ceck because I noticed it's allowed by the API to do `dev = BLEDevice(addr)`, as an alternative to getting a BLEDevice back from a scanner method. In that case, `details is None`. I can't think of a use case for that, but nevertheless it seems to be allowed by the API.

It might be worth considering disallowing use of the BLEDevice constructor in that way, if it's not a valid form for the object? (I can submit a PR for that change, if you'd like. Or can just leave it like it is now.)

Reply:
The `BLEDevice` constructor is not considered part of the public API. (and there are already some changes in the work for this in #1059)

So let's just leave it as-is for now.

Reply:
No worries. Appreciate the explanation!